tests/e2e/specs/layers.spec.js

```diff
@@ -300,6 +310,70 @@ describe('Test of layer handling', () => {
                 expect(visibleIds).to.contain(testLayerId)
             })
         })
+        it('keeps the layer settings when changing language', () => {
+            const langBefore = 'en'
+            const langAfter = 'de'
+            let activeLayersConfigBefore
+
+            // Change the layer configurations to non-default values.
+            cy.readWindowValue('store').then((store) => {
+                visibleLayerIds.forEach((layerId) => {
+                    store.dispatch('toggleLayerVisibility', { layerId })
+                    store.dispatch('setLayerOpacity', {
+                        layerId,
+                        opacity: 0.1,
+                    })
+                })
+                cy.fixture('layers.fixture.json').then((fixtures) => {
+                    const layerId = 'test.timeenabled.wmts.layer'
+                    const layerMetadata = fixtures[layerId]
+                    const layerTimestamp = getRandomTimestampFromSeries(layerMetadata)
+                    store.dispatch('setTimedLayerCurrentTimestamp', {
+                        layerId,
+                        timestamp: layerTimestamp,
+                    })
+                })
+            })
+
+            cy.readStoreValue('state').then((state) => {
+                // Check the language before the switch.
+                expect(state.i18n.lang).to.eq(langBefore)
+                state.layers.activeLayers
+                    .filter((layer) => 'lang' in layer)
+                    .forEach((layer) => expect(layer.lang).to.eq(langBefore))
+                // Save the layer configuration before the switch.
+                activeLayersConfigBefore = JSON.stringify(
+                    state.layers.activeLayers,
+                    stringifyWithoutLang
+                )
+            })
+
+            // Wait until the layers have been updated. We have to wait here
+            // to make sure the next time we wait we catch the language change.
+            cy.wait('@layers-configured')
+
+            // Change the language to a non-default value.
+            cy.get('[data-cy="menu-settings-section"]').click()
+            cy.get(`#menu-lang-selector button:contains("${langAfter.toUpperCase()}")`).click()
+
+            // Wait until the layers have been updated. This happens after
+            // the @layers have been loaded and the @topics are also too fast.
+            cy.wait('@layers-configured')
```

Comment:
IMO you could place this line directly in the helper function `goToMapView`, so that all tests uses it, but I'm not sure it will be needed anymore if you change the code as described in my comment above

Reply:
I moved the first `wait` to the `goToMapView` command.

Unfortunately, I couldn't remove either of the two. It looked good at first but during some runs the first `@layers-configured` was triggered too late and the second `wait` caught the first event. Removing the second `wait` also passed during some runs but less often.